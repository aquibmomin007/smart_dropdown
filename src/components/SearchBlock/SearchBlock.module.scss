.searchBlock{
    display: flex;
    flex-flow: row;
    padding: 5px 10px 10px;
    justify-content: space-between;
    align-items: center;
}

.searchBlockText{
    margin: 0;
    font-style: italic;
    font-size: 13px;
    color: #525050;
    word-break: break-word;
    padding: 0 10px 0 0;
    box-sizing: border-box;
}

.searchBlockBtn{
    line-height: 18px;
    padding: 5px;
    box-sizing: border-box;
    cursor: pointer;
    min-width: 100px;
}

.dropdownSearchInput {
    border: 1px solid #ccc;
    padding: 6px;
    line-height: 18px;
    width: calc(100% - 16px);
}

.searchInputWrapper{
    position: relative;
    padding: 15px 9px 10px;
}

.searchInputSpinnerWrapper{
    position: absolute;
    right: 13px;
    top: 19px;
    background: white;
    padding: 4px;
    z-index: 999;
    height: 16px;
}

.searchInputSpinner{
    -webkit-animation:spin 2s linear infinite;
    -moz-animation:spin 2s linear infinite;
    animation:spin 2s linear infinite;
}

@-moz-keyframes spin { 100% { -moz-transform: rotate(360deg); } }
@-webkit-keyframes spin { 100% { -webkit-transform: rotate(360deg); } }
@keyframes spin { 100% { -webkit-transform: rotate(360deg); transform:rotate(360deg); } }