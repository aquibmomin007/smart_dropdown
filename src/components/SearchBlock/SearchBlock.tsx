import React, {  } from 'react';
import styles from './SearchBlock.module.scss'

type SearchBlockProps = {
  searchTerm: string;
  hasAddPermission: boolean;
  handleAddNewOption: () => void;
  handleSearchChange: (event: any) => void;
  showSearchAddBlock: () => boolean
}

const SearchBlock = ({ searchTerm, hasAddPermission, handleAddNewOption, handleSearchChange, showSearchAddBlock }: SearchBlockProps) => { 
  return (
    <>
    <input 
      type="text" 
      value={searchTerm} 
      className={styles.dropdownSearchInput} 
      placeholder="Search..." 
      onChange={handleSearchChange} 
    />
    {showSearchAddBlock() && (
      <div className={styles.searchBlock}>
        <p className={styles.searchBlockText}>{`"${searchTerm}" not found`}</p>
        {hasAddPermission && <button className={styles.searchBlockBtn} onClick={handleAddNewOption}>Add &amp; Select</button>}
      </div>
    )}
    </>
  )
}

export default SearchBlock;