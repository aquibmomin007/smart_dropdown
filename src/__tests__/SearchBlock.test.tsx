import React from 'react';
import SearchBlock, {SearchBlockProps} from '../components/SearchBlock/SearchBlock';
import { render } from '@testing-library/react';


describe('SearchBlock Testing', () => {
  test('renders correctly and Add Button is shown', () => {
    const searchBlockProps:SearchBlockProps = {
      searchTerm: '',
      hasAddPermission: true,
      handleAddNewOption: jest.fn(),
      handleSearchChange: jest.fn(),
      showSearchAddBlock: jest.fn().mockReturnValue(true),
    };

    const { container, getByTestId, debug } = render(
        <SearchBlock {...searchBlockProps} />
    )
    const searchAddBtnHtml = getByTestId('add-block-test').querySelector(
      'button'
    ) as HTMLInputElement;
    expect(container).toMatchSnapshot();
    expect(searchAddBtnHtml).not.toBeNull()
  });

  test('renders correctly and Add Button is not shown', () => {
    const searchBlockProps:SearchBlockProps = {
      searchTerm: '',
      hasAddPermission: false,
      handleAddNewOption: jest.fn(),
      handleSearchChange: jest.fn(),
      showSearchAddBlock: jest.fn().mockReturnValue(true),
    };

    const { container, getByTestId, debug } = render(
        <SearchBlock {...searchBlockProps} />
    )
    const searchAddBtnHtml = getByTestId('add-block-test').querySelector(
      'button'
    ) as HTMLInputElement;
    expect(container).toMatchSnapshot();
    expect(searchAddBtnHtml).toBeNull()
  });
});
