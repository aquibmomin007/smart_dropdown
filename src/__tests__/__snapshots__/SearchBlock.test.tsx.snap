// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBlock Testing renders correctly and Add Button is not shown 1`] = `
<div>
  <input
    class="dropdownSearchInput"
    placeholder="Search..."
    type="text"
    value=""
  />
  <div
    class="searchBlock"
    data-testid="add-block-test"
  >
    <p
      class="searchBlockText"
    >
      "" not found
    </p>
  </div>
</div>
`;

exports[`SearchBlock Testing renders correctly and Add Button is shown 1`] = `
<div>
  <input
    class="dropdownSearchInput"
    placeholder="Search..."
    type="text"
    value=""
  />
  <div
    class="searchBlock"
    data-testid="add-block-test"
  >
    <p
      class="searchBlockText"
    >
      "" not found
    </p>
    <button
      class="searchBlockBtn"
    >
      Add & Select
    </button>
  </div>
</div>
`;
